config/types/config.go

```diff
@@ -15,14 +15,20 @@
 package types
 
 import (
+	"errors"
 	"net/url"
 
 	ignTypes "github.com/coreos/ignition/config/v2_2/types"
 	"github.com/coreos/ignition/config/validate/astnode"
 	"github.com/coreos/ignition/config/validate/report"
 )
 
+var (
+	ErrInvalidVersion = errors.New("Invalid version. Only version 0 is supported")
+)
+
 type Config struct {
+	Version   int        `yaml:"version"`
```

Comment:
Is there any reason we'd want this to be a semver?

Probably worth noting in a comment that `version` is intentionally undocumented unless and until we need a version `1`.

Reply:
I don't think a semver is necessary and it muddles the json parsing a little. Given we don't plan to use it unless we make breaking changes, any minor (i.e. backwards compatible) changes wont need to bump it. Using semver would imply that we should be bumping it.

I'll add a comment though.